daemon/api_model.go

```diff
@@ -56,12 +57,28 @@ type postModelData struct {
 }
 
 func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
-	defer r.Body.Close()
+	contentType := r.Header.Get("Content-Type")
+	mediaType, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		// assume json body, as type was not enforced in the past
+		mediaType = "application/json"
+	}
+
+	switch mediaType {
+	case "application/json":
+		return storeRemodel(c, r)
+	default:
+		return BadRequest("unexpected media type %q", mediaType)
```

Comment:
should we have some "not implemented error" instead for the multipart types we now expect?

Reply:
Added